api/core/utils/detailed_schema_validation_test.py

```diff
@@ -0,0 +1,355 @@
+import pytest
+
+from core.domain.errors import JSONSchemaValidationError
+from core.utils.detailed_schema_validation import DetailedSchemaValidator, validate_with_detailed_errors
+
+
+class TestDetailedSchemaValidator:
+    def test_valid_data_passes(self):
+        """Test that valid data passes validation without error."""
+        schema = {
+            "type": "object",
+            "properties": {
+                "name": {"type": "string"},
+                "age": {"type": "integer", "minimum": 0},
+            },
+            "required": ["name"],
+        }
+        data = {"name": "John", "age": 30}
+
+        validator = DetailedSchemaValidator(schema)
+        # Should not raise any exception
+        validator.validate_with_detailed_errors(data)
+
+    def test_missing_required_field_single(self):
+        """Test error message for single missing required field."""
+        schema = {
+            "type": "object",
+            "properties": {
+                "name": {"type": "string"},
+                "age": {"type": "integer"},
+            },
+            "required": ["name"],
+        }
+        data = {"age": 30}
+
+        validator = DetailedSchemaValidator(schema)
+        with pytest.raises(JSONSchemaValidationError) as exc_info:
+            validator.validate_with_detailed_errors(data)
+
+        error_msg = str(exc_info.value)
+        assert "Missing required field 'name'" in error_msg
```

Comment:
Not sure that's sufficient. That would likely be true for a vanilla error message from the json_schema package.